src/ripple/ledger/tests/SkipList_test.cpp

```diff
@@ -0,0 +1,106 @@
+//-----------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2012, 2015 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <BeastConfig.h>
+#include <ripple/app/ledger/Ledger.h>
+#include <ripple/ledger/View.h>
+#include <ripple/basics/Log.h>
+#include <beast/unit_test/suite.h>
+
+namespace ripple {
+namespace test {
+
+class SkipList_test : public beast::unit_test::suite
+{
+    void
+    testSkipList()
+    {
+        Config const config;
+        beast::Journal const j;
+        std::vector<std::shared_ptr<Ledger>> history;
+        auto prev =
+            std::make_shared<Ledger>(create_genesis, config);
```

Comment:
Nitpick: `prev` could be scoped so it's not left dangling in the rest of the test.